app/models/custom_field.rb

```diff
@@ -70,6 +72,10 @@ def self.loan_questions(field_set = nil)
       { internal_name: field_set })
   end
 
+  def self.sort_by_required(loan)
+    all.sort_by { |i| [i.required_for?(loan) ? 0 : 1, i.position] }
```

Comment:
I mean right now this is not chainable, doesn't return an AR::Relation.
Let's discuss

On Tue, Jul 26, 2016 at 5:18 PM, Adam Konner <notifications@github.com>
wrote:

> In app/models/custom_field.rb
> <https://github.com/sassafrastech/madeline_system/pull/119#discussion_r72337975>
> :
>
> > @@ -70,6 +72,10 @@ def self.loan_questions(field_set = nil)
> >        { internal_name: field_set })
> >    end
> >
> > +  def self.sort_by_required(loan)
> > +    all.sort_by { |i| [i.required_for?(loan) ? 0 : 1, i.position] }
>
> It could just as well be. I prefer class methods to scopes because they
> have exactly the same effect and I don't have to remember another syntax. I
> read that recommended somewhere—I think in Rails' own documentation.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/119/files/2637ef77e0be95ee6eb88af261c4de88538cbb06#r72337975>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cOAc9dT3BgLpCeF1zPaACwMJWQ0Dks5qZnmTgaJpZM4JTMeF>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org